src/go/cmd/http-relay-server/server/broker.go

```diff
@@ -282,13 +300,16 @@ func (r *broker) SendResponse(resp *pb.HttpResponse) error {
 func (r *broker) ReapInactiveRequests(threshold time.Time) {
 	r.m.Lock()
 	for id, pr := range r.resp {
+		pr.sendMutex.Lock()
 		if pr.lastActivity.Before(threshold) {
 			slog.Info("Timeout on inactive request", slog.String("ID", id))
 			defer close(pr.requestStream)
+			close(pr.markReap)
 			defer close(pr.responseStream)
```

Comment:
yeah, I am confused by the defer as well, in fact with the defer, we close the channel after we unlock the mutexes. The only reason it does not panic is because it also deletes the entry from `r.resp`, so `SendResponse` couldn't get a reference to a channel that is about to be closed.

Reply:
I haven't reviewed the rest of the PR yet - but should we just try a separate PR to remove the defer and see if anything goes wrong?

Reply:
I went ahead and removed the defer in this PR, so far it hasn't impact any of the tests.